types/index.d.ts

```diff
@@ -0,0 +1,496 @@
+
+    /** Updates one or more records in the collection. */
+    update(target: object | number | string, attrs?: object): any;
+
+    /** Returns an array of models from `collection` that match the key-value pairs in the `query` object. */
+    where(query: object): any;
+  }
+}
+
+declare module "miragejs/identity-manager" {
+  /** An IdentityManager is a class that's responsible for generating unique identifiers. You can define a custom identity manager for your entire application, as well as on a per-model basis. */
+  export default class IdentityManager {
+    constructor();
+
+    get(): number;
+
+    /** Registers `uniqueIdentifier` as used. */
+    set(uniqueIdentifier: string | number): void;
+
+    inc(): number;
+
+    /**  Returns the next unique identifier. */
+    fetch(): string;
+
+    /** Resets the identity manager, marking all unique identifiers as available. */
+    reset(): void;
+  }
+}
+
+declare module "miragejs/orm/schema" {
+  import { Collection, ModelInstance } from "miragejs";
+  import Db from "miragejs/db";
+
+  type ModelInitializer<Data> = {
+    [K in keyof Data]: Data[K] extends Collection<infer M>
+      ? Collection<M> | M[]
+      : Data[K];
+  };
+
+  /**
+   * An interface to the Mirage ORM that allows for querying and creating records.
+   */
+  export default class Schema<Registry = Record<string, ModelInstance>> {
+    /** Mirage's in-memory database */
+    readonly db: Db;
+
+    /**
+     * Creates a model of the given type.
+     * @param modelName The type of model to instantiate
+     * @param data Optional initial values for model attributes/relationships
+     */
+    create<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<ModelInitializer<Init>>
+    >(
+      modelName: K,
+      data?: Data
+    ): Init & { [K in keyof Init & keyof Data]: Exclude<Init[K], undefined> };
+
+    /** Locates one or more existing models of the given type by ID(s). */
+    find<K extends keyof Registry>(type: K, id: string): Registry[K] | null;
+    find<K extends keyof Registry>(
+      type: K,
+      ids: string[]
+    ): Collection<Registry[K]>;
+
+    /** Locates an existing model of the given type by attribute value(s), if one exists. */
+    findBy<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Registry[K] | null;
+
+    /** Locates an existing model of the given type by attribute value(s), creating one if it doesn't exist. */
+    findOrCreateBy<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Registry[K];
+
+    /** Locates an existing model of the given type by attribute value(s), if one exists. */
+    where<K extends keyof Registry>(
+      type: K,
+      attributes: Partial<Registry[K]>
+    ): Collection<Registry[K]>;
+    where<K extends keyof Registry>(
+      type: K,
+      // tslint:disable-next-line:unified-signatures
```

Comment:
Is this `tslint:disable-next-line` actually required?

Reply:
The overload can be merged with the first declaration. Update committed.